src/components/members/hooks/useBlockMember.ts

```diff
@@ -0,0 +1,53 @@
+import { colors } from '@sopt-makers/colors';
+import { useToast } from '@sopt-makers/ui';
+import { useQueryClient } from '@tanstack/react-query';
+import { useRouter } from 'next/router';
+import { playgroundLink } from 'playground-common/export';
+
+import { usePostBlockMemberMutation } from '@/api/endpoint/members/postBlockMember';
+import useConfirm from '@/components/common/Modal/useConfirm';
+
+export const useBlockMember = () => {
+  const router = useRouter();
+  const queryClient = useQueryClient();
+
+  const { confirm } = useConfirm();
+  const { open } = useToast();
+  const { mutate } = usePostBlockMemberMutation();
+
+  const handleBlockMember = async (memberId?: number) => {
+    if (!memberId) throw new Error('Invalid Member id');
+
+    const result = await confirm({
+      title: '이 멤버를 차단하시겠습니까?',
+      description:
+        '차단 시 멤버 리스트는 유지되지만, 이 멤버의 게시글과 댓글이 보이지 않게 돼요. 한 번 차단한 멤버는 다시 해제할 수 없어요.',
+      okButtonText: '차단하기',
+      okButtonColor: colors.error,
+      okButtonTextColor: colors.white,
+      cancelButtonText: '취소',
+      maxWidth: 400,
+    });
+
+    if (result) {
+      mutate(
+        { blockedMemberId: memberId },
+        {
+          onSuccess: async () => {
+            queryClient.invalidateQueries({
+              predicate: (query) => query.queryKey.some((key) => String(key).includes('community')),
+            });
+            await router.push(playgroundLink.memberList());
```

Comment:
그냥 확실하게 다 날려버리는건 어때요? 그러면 API에 community 가 들어가지 않는 것에 대해 업데이트 되지 않는 것을 피하면서도, 차단하는 상황은 자주 일어나는 일은 아닐거라 서버의 부담이 크지도 않을 것 같아요.

```suggestion
            await queryClient.invalidateQueries();
            await router.push(playgroundLink.memberList());
```

Reply:
차단하는 상황이 자주 일어나는 일이 아니라서 다 날리는 것도 좋은 것 같네요!

Reply:
감사합니다! 반영했어요!